lib/oper.g

```diff
@@ -615,6 +615,92 @@ BIND_GLOBAL( "TraceImmediateMethods", function( arg )
     fi;
 end );
 
+#############################################################################
+##
+##
+##  <#GAPDoc Label="TraceInternalMethods">
+##  <ManSection>
+##  <Func Name="TraceInternalMethods" Arg=''/>
+##  <Func Name="UntraceInternalMethods" Arg=''/>
+##  <Func Name="GetTracedInternalMethodsCounts" Arg=''/>
+##  <Func Name="ClearTraceInternalMethodsCounts" Arg=''/>
+##
+##  <Description>
+##  <Ref Func="TraceInternalMethods"/> enables tracing for all internal methods.
+##  Internal methods are methods which implement many fundamental operations in GAP.
+##  In this version of GAP, the internal methods which can be traced is:
+##  <List>
+##  <Mark>Zero, ZeroMut</Mark><Item>Mutable and Immutable <Ref Attr="Zero"/></Item>
+##  <Mark>AInv, AInvMut</Mark><Item>Mutable and Immutable <Ref Attr="AdditiveInverse"/></Item>
+##  <Mark>One, OneMut</Mark><Item>Mutable and Immutable <Ref Attr="One"/></Item>
+##  <Mark>Inv, InvMut</Mark><Item>Mutable and Immutable <Ref Attr="Inverse"/></Item>
+##  <Mark>Sum</Mark><Item>The operator <Ref Oper="\+"/></Item>
+##  <Mark>Diff</Mark><Item>The operator <C>-</C> operator</Item>
+##  <Mark>Prod</Mark><Item>The operator <Ref Oper="\*"/></Item>
+##  <Mark>Quo</Mark><Item>The operator <Ref Oper="\/"/></Item>
+##  <Mark>LQuo</Mark><Item>The left-quotient operator</Item>
+##  <Mark>Pow</Mark><Item>The operator <Ref Oper="\^"/></Item>
+##  <Mark>Comm</Mark><Item>The operator <Ref Oper="Comm"/></Item>
+##  <Mark>Mod</Mark><Item>The operator <Ref Oper="\mod"/></Item>
+##  </List>
+##  <P/>
+##  <Ref Func="UntraceInternalMethods"/> turns tracing off.
+##  As these methods can be called hundreds of thousands of times in simple GAP
+##  code, there isn't a statement printed each time one is called. Instead, the
+##  method <Ref Func="GetTracedInternalMethodsCounts"/> returns how many times
+##  each operation has been applied to each type of variable (the type of a
+##  variable can be found with the <C>TNAM_OBJ</C> method).
+##  The return value for two argument operators is a record of records <C>r</C>, where
+##  <C>r.op</C> stores information about operator <C>op</C>. For one argument operators
+##  <C>r.op.i</C> stores how many times <C>op</C> was called with an argument of type
+##  <C>i</C>, while for two argument operators <C>r.op.i.j</C> stores how many times
+##  <C>op</C> was called with arguments of type <C>i</C> and <C>j</C>.
+##  <Log><![CDATA[
+## gap> TraceInternalMethods();
+## true
+## gap> 2+3+4+5+6;;
+## gap> 2.0+2.0;;
+## gap> 3^(1,2,3);;
+## gap> GetTracedInternalMethodsCounts();
+## rec( Pow := rec( integer := rec( ("permutation (small)") := 1 ) ),
+##      Sum := rec( integer := rec( integer := 4 ), macfloat := rec( macfloat := 1 ) ) )
```

Comment:
In the compiled documentation, the last `)` is outside of the rectangular box in which Examples are printed. Thus, it looks like a overfull hbox. Maybe you could insert another line break?

Reply:
If this is tested by Travis etc, then Chris doesn't have any choice about how the output should look; if he adds arbitrary line breaks, then the tests will fail when the manual examples are tested. However I'm not sure where this _is_ ever tested by Travis.

Reply:
Oh ok, I see. I guess I won't take the risk of a manual formatting now because it might be tested in the future and thing could break?

Reply:
THis formatting was actually done by me, so I'll shorten/neaten it